tools/harvest_source_import/remote_ckan/lib.py

```diff
@@ -194,34 +174,77 @@ def create_organization(self, data):
 
         # TODO get the organization_type GSA field
 
+        return self.request_ckan(method='POST', url=org_create_url, data=organization)
+         
+    def request_ckan(self, method, url, data):
+        """ request CKAN and get results """
+
+        headers = self.get_request_headers(include_api_key=True)
+
         try:
-            req = requests.post(package_create_url, data=organization, headers=headers)
+            if method == 'POST':
+                req = requests.post(url, data=data, headers=headers)
+            elif method == 'GET':
+                req = requests.get(url, params=data, headers=headers)
+            else:
+                raise ValueError(f'Invalid method {method}')
+
         except Exception as e:
-            error = 'ERROR creating organization: {} [{}]'.format(e, organization)
-            self.errors.append(error)
+            error = 'ERROR at {} {}: {}'.format(url, data, e)
             return False, 0, error
 
         content = req.content
         try:
             json_content = json.loads(content)
         except Exception as e:
-            error = 'ERROR parsing JSON data: {} [{}]'.format(content, e)
+            error = 'ERROR parsing JSON: {} {}: {}'.format(url, content, e)
             logger.error(error)
             return False, 0, error
         
         if req.status_code >= 400:
             if 'already in use' in str(content):
                 return False, req.status_code, 'Already exists'
-            error = ('ERROR creating organization: {}'
-                     '\n\t Status code: {}'
-                     '\n\t content:{}'.format(organization, req.status_code, req.content))
+            error = ('ERROR status: {}'
+                     '\n\t content:{}'.format(req.status_code, req.content))
+            self.errors.append(error)
             logger.error(error)
             return False, req.status_code, error
 
         if not json_content['success']:
             error = 'API response failed: {}'.format(json_content.get('error', None))
             logger.error(error)
+            self.errors.append(error)
             return False, req.status_code, error
 
-        logger.info('Organization created OK {}'.format(organization['title']))
-        return True, req.status_code, None
\ No newline at end of file
+        logger.info('Request OK {}'.format(url))
+        return True, req.status_code, None
+    
+    def get_config(self, data):
+        """ get config and extras from full data package and return a final str config """
+        config = data.get('config', {})
+        if type(config) == str:
+            config = json.loads(config)
+
+        # we can also have config in extras
+        # we don't get extras 
```

Comment:
Can you clarify the comment above? Maybe "We may not have extras in every response"?

Reply:
Fixed